roofit/roofitcore/src/RooArgSet.cxx

```diff
@@ -649,3 +649,32 @@ Bool_t RooArgSet::isInRange(const char* rangeSpec)
 
 
 void RooArgSet::processArg(double value) { processArg(RooFit::RooConst(value)); }
+
+
+#include "TGlobal.h"
+#include "TROOT.h"
+#include "TSystem.h"
+
+bool RooArgSet::pyRootInitialized = false;
+
+namespace {
+
+// Helper function to create a RooArgSet from a PyObject, using the interpreter to avoid dependency of RooFit on pyROOT.
+std::vector<RooAbsCollection::RooAbsArgPtrOrDouble> const& rooAbsArgPtrOrDoubleFromPyObject(PyObject * pyobject) {
+  if(!RooArgSet::pyRootInitialized) {
+    gROOT->ProcessLine("namespace RooFit { namespace Detail { std::vector<RooAbsArgPtrOrDouble> rooAbsArgPtrOrDoubleFromPyObject(PyObject *pyobject); }};");
+    gROOT->ProcessLine("std::vector<RooAbsCollection::RooAbsArgPtrOrDouble> __rooAbsArgPtrOrDoubleTmp;");
+    RooArgSet::pyRootInitialized = true;
+  }
+  std::stringstream ss;
+  ss << "__rooAbsArgPtrOrDoubleTmp = RooFit::Detail::rooAbsArgPtrOrDoubleFromPyObject(reinterpret_cast<PyObject*>("
+     << reinterpret_cast<uintptr_t>(pyobject) << "));";
+  gROOT->ProcessLine(ss.str().c_str());
+  return *reinterpret_cast<std::vector<RooAbsCollection::RooAbsArgPtrOrDouble>*>(gROOT->GetGlobal("__rooAbsArgPtrOrDoubleTmp")->GetAddress());
+}
+
+} // namespace
+
+/// RooArgSet constructor from a Python object, will throw if the object is not
+/// a Python set containing only RooAbsArgs and floats.
+RooArgSet::RooArgSet(PyObject * pyobject) : RooArgSet{rooAbsArgPtrOrDoubleFromPyObject(pyobject)} {}
```

Comment:
I guess the same can't be done in Python by pythonizing `__init__`, is that correct? Since the C++ constructor will be called directly during overload resolution.

Reply:
Yes, that's correct. I tried it out first by pythonizing `RooArgSet.__init__` but it didn't work.